starter_code/src/movies.js

```diff
@@ -1,20 +1,150 @@
-/* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
+function turnHoursToMinutes (moviesArray){
+  if (!moviesArray) return;
 
+  return moviesArray.map(function(movie){
+    var newMovie = Object.assign({}, movie);
+    var movieDuration = newMovie.duration.split(" ");
+    
+    if (movieDuration.length === 2) {
+      movieDuration[0] =  Number(movieDuration[0].replace("h", "")) * 60;
+      movieDuration[1] = Number(movieDuration[1].replace("min", ""));
+    }
+    else{
+      if (movieDuration[0].indexOf("h") === -1) {
+        movieDuration[0] = Number(movieDuration[0].replace("min", ""));
+      }
+      else{
+        movieDuration[0] = Number(movieDuration[0].replace("h", "")) * 60;
+      }
+    }    
 
-// Get the average of all rates with 2 decimals 
+    newMovie.duration = movieDuration.reduce(function(acc, duration){
+      return acc + duration;
+    }, 0);
 
+    return newMovie;
+  })
+}
+  
+  
+// Get the average of all rates with 2 decimals 
+function ratesAverage(moviesArray) {
+  if (!moviesArray) return;
 
+  return moviesArray.reduce(function(acc, movie, index, moviesArray){
+    if (index === moviesArray.length - 1) {
+      return Number(((Number(acc) + Number(movie.rate)) / moviesArray.length).toFixed(2));
+    } else {
+      return Number((Number(acc) + Number(movie.rate)).toFixed(2));
+    }
+  }, 0);
+}
+  
 // Get the average of Drama Movies
+function dramaMoviesRate(moviesArray) {
+  if (!moviesArray) return;
+
+  if (moviesArray.filter(function(movie){ return movie.genre.indexOf("Drama") !== -1; }).length === 0)
+    return;
 
+  return moviesArray
+    .filter(function(movie){
+      return movie.genre.indexOf("Drama")!==-1;
+    })
+    .reduce(function(acc, movie, index, moviesArray){
+      if (index === moviesArray.length - 1) {
+        return Number(((Number(acc) + Number(movie.rate)) / moviesArray.length).toFixed(2));
+      } else {
+        return Number((Number(acc) + Number(movie.rate)).toFixed(2));
+      }
+    }, 0);
+}
 
 // Order by time duration, in growing order
+function orderByDuration(moviesArray){
+  if (!moviesArray) return;
+
```

Comment:
El ejercicio no os pide, en 'orderByDuration', devolver undefined si el array está vacío o si no existe. No pongáis cosas por poner si no hacen falta